plugins/pulp_rpm/plugins/profilers/yum.py

```diff
@@ -287,8 +287,9 @@ def _get_rpms_from_errata(errata):
         """
         rpms = []
         if "pkglist" not in errata.metadata:
-            _logger.warning(_("metadata for errata <%s> lacks a 'pkglist'") % (
-                            errata.unit_key.get('errata_id')))
+            msg = _("metadata for errata <%(errata_id)s> lacks a 'pkglist'")
+            msg_dict = {'errata_id': errata.unit_key.get('errata_id')}
+            _logger.warning(msg % msg_dict)
```

Comment:
Consider `_logger.warning(msg, msg_dict)` instead of using `%`.  This way the logger (`warning()` method in this case) will do formatting and variable substitution only when the logging level criteria is met.  

This applies to all logging calls.